lib/ansible/plugins/lookup/cyberarkpassword.py

```diff
@@ -0,0 +1,142 @@
+# (c) 2017, Edward Nunez <edward.nunez@cyberark.com>
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+import os
+import subprocess
+
+from ansible.errors import AnsibleError
+from ansible.plugins.lookup import LookupBase
+
+CLIPASSWORDSDK_CMD = '/opt/CARKaim/sdk/clipasswordsdk'
+if os.getenv('AIM_CLIPASSWORDSDK_CMD') is not None:
+    CLIPASSWORDSDK_CMD = os.environ['AIM_CLIPASSWORDSDK_CMD']
+
+
+class CyberarkPassword:
+
+    def __init__(self, **kwargs):
+
+        self.appid = kwargs.pop('appid', None)
+        self.query = kwargs.pop('query', None)
+        self.output = kwargs.pop('output', None)
+
+        # Support for Generic parameters to be able to specify
+        # FailRequestOnPasswordChange, Queryformat, Reason, etc.
+        self.extraParms = []
+        for key, value in kwargs.items():
+            self.extraParms.append('-p')
+            self.extraParms.append(key + "=" + value)
+
+        if self.appid is None:
+            raise AnsibleError("No Application ID specified")
+        if self.query is None:
+            raise AnsibleError("No Vault query specified")
```

Comment:
It may be beneficial to clarify that this is cyberark vault, to avoid confusion where someone may think the error is in reference to ansible vault.